### ISSUE TITLE
`const_source_location` returning empty values where MRI does not

### ISSUE THREAD

**Issue by nirvdrum** [oracle/truffleruby#2833]
I'm running into an issue booting [Sorbet](https://github.com/sorbet/sorbet). It uses `const_source_location` to build up [a path to the lib directory](https://github.com/sorbet/sorbet/blob/1204e682b8e83a919d8ec8a0753022f241f393ee/gems/sorbet-runtime/lib/types/private/methods/_methods.rb#L122). This works fine on MRI as it returns the filename defining the module whereas TruffleRuby returns `[]`.

I've narrowed down the TruffleRuby issue to a simple reproduction (placed in a file named _bad.rb_):

```ruby
module T
  module Private
  end
end

puts T.const_source_location(:Private)
```

On MRI, I get:

```
> ruby -v bad.rb
ruby 3.1.3p185 (2022-11-24 revision 1a6b16756e) [arm64-darwin21]
bad.rb
2
```

On TruffleRuby, I get:

```
> jt ruby -v bad.rb
truffleruby 23.0.0-dev-71f07786*, like ruby 3.1.3, GraalVM CE JVM [aarch64-darwin]
```

I've looked at the issue briefly and it's due to the constants not having associated source sections. In that case, we have `const_source_location` return `[]`.

**Comment by eregon** [comment]
This is an easy fix (#2835), interestingly we already recorded the SourceSection on RubyModule itself for interop, which is something Ruby doesn't have a way to query, but we missed to pass a node/SourceSection to `setConstantInternal()`.

**Comment by nirvdrum** [comment]
Great. I was planning to pick this one up. I was hoping it was easy, but feared it might require parser changes.
